moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -538,6 +538,12 @@ bool Trajectory::integrateForward(std::list<TrajectoryStep>& trajectory, double
 
     if (next_discontinuity != switching_points.end() && path_pos > next_discontinuity->first)
     {
+      // Avoid having a TrajectoryStep with path_pos near a switching point which will cause an almost identical
+      // TrajectoryStep get added in the next run (https://github.com/ros-planning/moveit/issues/1665)
+      if (path_pos - next_discontinuity->first < time_step_ * 1e-5)
```

Comment:
Is it necessary to allow the trajectory steps to be much shorter than the time step?   I worry slightly that this leaves the possibility for very rare errors.

Reply:
It seems that even with a timestep of 1e-8 an change of velocity of 1 will give an an acceleration of 100,000,000. Maybe when we eventually divide change in velocity by time-step, we cap the result to some maximum value?

Reply:
This should be a parameter then so this can be adjusted to the corresponding setup. I'm afraid this would be a magic parameter, then

Reply:
We could use EPS